features/0577-self-encoded-keys/README.md

```diff
@@ -0,0 +1,296 @@
+
+**Ed25519 (Authcrypt):**
+
+The algorithm for usage of Ed25519 keys identified in RFC0019 for authenticated encryption is `Authcrypt`.
+
+The multicodec identifier for Ed25519 public keys is `ed25519-pub` therefore its code is `0xed`.
+
+An example output from a function implementing the [syntax](#syntax) would be: `Authcrypt|z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH`.
+
+**X25519:**
+
+Several algorithms are identified for use with X25519 keys as described in [RFC7518](https://tools.ietf.org/html/rfc7518#section-4.6) and registered the JOSE registry; this example will use `ECDH-ES+A256KW`.
+
+The multicodec identifier for X25519 public keys is `x25519-pub` and its code is `0xec`.
+
+An example output from a function implementing the [syntax](#syntax) would be: `ECDH-ES+A256KW|zBzoR5sqFgi6q3iFia8JPNfENCpi7RNSTKF7XNXX96SBY4`.
+
+**P-256:**
+
+The algorithm for usage of P-256 keys along with the SHA-256 hash function identified in [RFC7518](https://tools.ietf.org/html/rfc7518#section-3.4) and registered in the JOSE registry is `ES256`.
+
+The multicodec identifier for P-256 public keys is `p256-pub` and its code is `0x1200`.
+
+An example output from a function implementing the [syntax](#syntax) would be: `ES256|z3u1vAS3rWXpB1icR5Df5yUaVtn2Q2i1AMCKxBACNNqsxJa1`
+
+
+**P-384:**
+
+The algorithm for usage of P-384 keys along with the SHA-384 hash function identified in [RFC7518](https://tools.ietf.org/html/rfc7518#section-3.4) and registered in the JOSE registry is `ES384`.
+
+The multicodec identifier for P-384 public keys is `p384-pub` and its code is `0x1201`.
+
+An example output from a function implementing the [syntax](#syntax) would be: `ES384|z2bHnTFkhn3NLWvF4qei6cLBiL5xtvuY8wt74jvi7w9rA3W29i6st9YkmL8EYBsbARTrrN`.
+
+## Changes to RFCs
+
+Should this RFC be [ACCEPTED](https://github.com/hyperledger/aries-rfcs/blob/master/README.md#accepted), a [community-coordinated update](https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0345-community-coordinated-update/README.md) will be used to apply updates to the agent code bases and impacted RFCs.
+
+### RFC0067 DIDComm DIDDoc Conventions
+
+* Switch to the new syntax for inlined keys
+* Separate authentication keys (for signatures) from key agreement keys (for encryption)
+
+This RFC proposes the addition of a new attribute to the [`service`](https://github.com/hyperledger/aries-rfcs/blob/master/features/0067-didcomm-diddoc-conventions/README.md#service-conventions) object: `authnKeys`.
+
+Example with all keys inlined:
+
+```diff
+{
+  "service": [{
+     "id": "did:example:123456789abcdefghi#did-communication",
+     "type": "did-communication",
+     "priority" : 0,
+-    "recipientKeys" : [ "did:example:123456789abcdefghi#1" ],
+-    "routingKeys" : [ "did:example:123456789abcdefghi#1" ],
++    "authnKeys": ["ES384|z2bHnTFkhn3NLWvF4qei6cLBiL5xtvuY8wt74jvi7w9rA3W29i6st9YkmL8EYBsbARTrrN"],
```

Comment:
Does this need to be ordered list that matches recipientKeys? E.g does the nth authnKeys entry correspond to the nth recipientKeys entry? If so, would it be better to make the ordering more explicit -- e.g. one array, with two items in each?

Reply:
`authnKeys` exists for a signature required in the did-exchange protocol. Ordering does not matter so long as the signature is from one of these keys.

Reply:
Cool - thanks.